tests.py

```diff
@@ -0,0 +1,303 @@
+                'baz1': [],
+                'baz2': None,
+            }
+        },
+        'foo3': {
+            'bar': {
+                'baz1': [],
+                'baz2': [],
+            }
+        }
+    }
+)
+def test_rmdir(client):
+    client.rmdir('/foo1/bar/')
+    assert not client.exists('/foo1/bar/')
+
+    pytest.raises(ValueError, client.rmdir, '/foo2/bar/baz2')
+
+    pytest.raises(RuntimeError, client.rmdir, '/foo2/bar/')
+    client.rmdir('/foo2/bar/', recursive=True)
+    assert not client.exists('/foo2/bar/')
+
+    pytest.raises(RuntimeError, client.rmdir, '/foo3/')
+    client.rmdir('/foo3/', recursive=True)
+    assert not client.exists('/foo3/')
+
+    client.rmdir('/', recursive=True)
+    assert not client.exists('/foo1/')
+
+
+@mkdirs_recursively(
+    tree={
+        'foo1': {
+            'bar': [],
+        },
+        'foo2': {
+            'bar': {
+                'baz1': [],
+                'baz2': None,
+            }
+        },
+        'foo3': {
+            'bar': {
+                'baz1': [],
+                'baz2': [],
+            }
+        }
+    }
+)
+def test_rm(client):
+    client.rm('/foo1/bar/')
+    assert not client.exists('/foo1/bar/')
+
+    client.rm('/foo2/bar/baz2')
+    assert not client.exists('/foo1/bar/baz2')
+
+    pytest.raises(RuntimeError, client.rm, '/foo2/bar/')
+    client.rm('/foo2/bar/', recursive=True)
+    assert not client.exists('/foo2/bar/')
+
+    pytest.raises(RuntimeError, client.rm, '/foo3/')
+    client.rm('/foo3/', recursive=True)
+    assert not client.exists('/foo3/')
+
+    client.rm('/', recursive=True)
+    assert not client.exists('/foo1/')
+
+
+@mkdirs_recursively(
+    tree={
+        'foo': {
+            'bar': None
+        }
+    }
+)
+def test_get_data(client):
+    file_obj = client.get('/foo/bar')
+    assert file_obj.name == 'bar'
+    assert file_obj.full_path == '/foo/bar'
+    assert file_obj.data == FILE_CONTENT
+
+
+@mkdirs_recursively(
+    tree={
+        'foo': {
+            'bar1': [],
+            'bar2': {
+                'baz': [],
+            },
+        }
+    }
+)
+def test_get_folder(client):
+    folder_obj1 = client.get('/foo/bar1/')
+    assert folder_obj1.name == 'bar1/'
+    assert folder_obj1.metadata.is_dir
+
+    folder_obj2 = client.get('/foo/bar2/baz/')
+    assert folder_obj2.name == 'baz/'
+    assert folder_obj2.metadata.is_dir
+
+
+@pytest.mark.skip()
+@mkdirs_recursively(
+    tree={
+        'foo': {
+            'bar1': [],
+            'baz': None,
+            'bar2': [],
+        }
+    }
+)
+def test_cp(client):
+    client.cp('/foo/baz', '/foo/bar2/')
+    assert client.exists('/foo/bar2/baz')
+
+    client.cp('/foo/bar2/', '/foo/bar1/')
+    assert client.exists('/foo/bar1/bar2/baz')
+
+
+@mkdirs_recursively(
+    tree={
+        'foo': {
+            'bar1': None,
+            'bar2': None,
+            'bar3': None,
+        },
+        'baz': [],
+    }
+)
+def test_get_last_object(client):
```

Comment:
add test for last object where there are timezone differences

Reply:
not necessary, using minio's timezone